src/main/tut/3.2-adjunctions.md

```diff
@@ -0,0 +1,130 @@
+```Haskell
+swap       :: (a,b) -> (b,a)
+swap (a,b) = (b,a)
+```
+```tut:silent
+def swap[A, B]: ((A, B)) => (B, A) = {
+  case (a, b) => (b, a)
+}
+```
+................
+```Haskell
+return :: d -> m d
+```
+```scala
+def `return`[D]: D => M[D]
```

Comment:
an `M[_]` type parameter is missing here

Reply:
> Here, m is the (endo-) functor corresponding to R∘L, and w is the (endo-) functor corresponding to L∘R. As we’ll see later, they are part of the definition of a monad and a comonad, respectively.

It seemed to me like kind of a concrete example, however simplified:

```scala
 trait Extract[M[_]] { // ingnore
  def return[D](d: D): M[D]
} //ignore
```